app/controllers/vendors_controller.rb

```diff
@@ -0,0 +1,66 @@
+class VendorsController < ApplicationController
+  # (30*24*60*60) = 2592000
+  ONE_MONTH = 2592000
+
+  def index
+    @vendors = Vendor.all
+    @main_page = true
+  end
+
+  def show
+    @vendor = Vendor.find(params[:id])
+    @vendor ||= Vendor.find(params[:vendor_id])
```

Comment:
This is a little unclear-- why is @vendor being reassigned using the params[:vendor_id] here if it doesn't exist yet?